featuretools/entityset/entityset.py

```diff
@@ -1433,6 +1434,14 @@ def update_dataframe(self, dataframe_name, df, already_sorted=False, recalculate
             self[dataframe_name].ww.pop(last_time_index_column)
             del self[dataframe_name].ww.metadata['last_time_index']
 
+        # If the original DataFrame had an index created via make_index, we may need to remake the index
+        created_index = self[dataframe_name].ww.metadata.get('created_index')
+        if created_index is not None:
+            if created_index not in df.columns:
+                df = _create_index(df, created_index)
+            else:
+                warnings.warn(f'New DataFrame, {dataframe_name}, already has the created index column, {created_index}. ')
```

Comment:
Decided to allow users to pass in a DataFrame that has the index column present. Figure this could be important for instances where someone is making changes to the DataFrame that's currently in the EntitySet, since they wouldn't necessarily want their data re-indexed. I'm not sure that this warning is actually necessary, as it's warning on the lack of action being taken. 

Happy to change to an error, if this behavior seems confusing or unexpected, since we do raise an error at `add_dataframe` if `make_index` is true but the `index` column is already in the DataFrame.

Reply:
If we do make this error, we'll have to update `EntitySet.concat` which calls `replace_dataframe` on the concated dataframes, and should probably keep whatever index was created from the concat call and not reindex there.

Reply:
Removing the warning makes sense to me, as I did feel weird about raising a warning on our lack of action.